File: src/api/AMP.EMS.API/Core/Entities/EventGuestRole.cs
using AMP.Infrastructure.Entity;

namespace AMP.EMS.API.Core.Entities;

public class EventGuestRole : FullAuditableEntity<Guid>
{
    public Guid RoleId { get; set; }
    public Guid EventGuestId { get; set; }
    public Role? Role { get; set; }
    public EventGuest? EventGuest {get;set;}
}

File: src/ui/AMP.EMS/src/app/core/models/event-guest-role.ts
import { Event } from "./event";
import { EventGuest } from "./event-guest";
import { Role } from "./role";

export interface EventGuestRole {
    id?: string;
    roleId?: string;
    eventGuestId?: string;
    eventGuest?: EventGuest;
    role?: Role;
}
